lisa/tools/hugepages.py

```diff
@@ -0,0 +1,115 @@
+    @property
+    def command(self) -> str:
+        return ""
+
+    @property
+    def can_install(self) -> bool:
+        return False
+
+    def _initialize(self, *args: Any, **kwargs: Any) -> None:
+        self._hugepage_sizes: Set[int] = set()
+        self._hugepages_dir_path = "/sys/devices/system/node/node*/hugepages"
+
+    def get_hugepage_sizes_in_kb(self) -> None:
+        """
+        This function get the hugepage sizes in kB available on the system.
+        This is done by listing the hugepages directory
+        and extracting the size from the directory name.
+        e.g. [
+        "'/sys/devices/system/node/node0/hugepages/hugepages-16777216kB/'",
+        "'/sys/devices/system/node/node0/hugepages/hugepages-2048kB/'",
+        "'/sys/devices/system/node/node0/hugepages/hugepages-524288kB/'"
+        ]
+        """
+        ls = self.node.tools[Ls]
+        # if not ls.path_exists(self._hugepages_dir_path, sudo=True):
+        #     raise SkippedException(f"path: {self._hugepages_dir_path} does not exist")
+        hugepage_size_dir_names = ls.list_dir(self._hugepages_dir_path, sudo=True)
+        for hugepage_size_dir_name in hugepage_size_dir_names:
+            matched_hugepage = PATTERN_HUGEPAGE.match(hugepage_size_dir_name)
+            if not matched_hugepage:
+                raise LisaException(
+                    f"no hugepage size found in {hugepage_size_dir_name}"
+                )
+            self._hugepage_sizes.add(int(matched_hugepage.group("hugepage_size_in_kb")))
+
+    def _enable_hugepages(
+        self, hugepage_size_kb: HugePageSize = HugePageSize.HUGE_2MB
+    ) -> None:
+        echo = self.node.tools[Echo]
+        meminfo = self.node.tools[Free]
+        nics_count = len(self.node.nics.get_nic_names())
+        numa_nodes = self.node.tools[Lscpu].get_numa_node_count()
+
+        request_space_kb = (nics_count - 1) * 1024 * 1024 * numa_nodes * 2
+        free_memory_kb = meminfo.get_free_memory_kb()
+
+        if free_memory_kb < request_space_kb:
+            self.node.log.debug(
+                f"WARNING: Not enough {hugepage_size_kb.value} KB pages "
+                "available for DPDK! "
+                f"Requesting {request_space_kb} KB found {free_memory_kb} "
+                "KB free."
+                "Test may fail if it cannot allocate memory."
+            )
+        request_pages = request_space_kb // hugepage_size_kb.value
+        for i in range(numa_nodes):
+            # nr_hugepages will be written with the number calculated
+            # based on 2MB hugepages if not specified, subject to change
+            # this based on further discussion
+            echo.write_to_file(
+                f"{request_pages}",
+                self.node.get_pure_path(
+                    f"/sys/devices/system/node/node{i}/hugepages/"
+                    f"hugepages-{hugepage_size_kb.value}kB/nr_hugepages"
+                ),
+                sudo=True,
+            )
```

Comment:
Also; check the return code here. If the write failed or we didn't get the value we asked for; raise an exception.

Reply:
it doesn't return a code but it does perform an assert inside